sonar-core/src/main/java/org/sonar/core/issue/tracking/SimpleTracker.java

```diff
@@ -31,10 +31,10 @@
     Tracking<RAW, BASE> tracking = new Tracking<>(rawInput, baseInput);
 
     // 1. match issues with same rule, same line and same line hash, but not necessarily with same message
-    match(tracking, LineAndLineHashKeyFactory.INSTANCE, true);
+    match(tracking, LineAndLineHashKey::new, true);
```

Comment:
The factory is always the same. If you don't want to keep the INSTANCE, what about creating as a static member field?

Reply:
Arent't method references cheap, and designed to allow future optimization by the JVM, instead of creating an object?

Reply:
You're right, not a big deal